doc/ExpertOptions.md

```diff
@@ -1470,58 +1480,62 @@ You would write the following manifest:
 {
   "test.input_manifest___": {
     "s": "hello",
-    "f": "dx://file-xxx"
+    "f": "project-aaa:file-xxx"
   }
 }
 ```
 
-When you compile the workflow, provide the manifest using the `-inputs` option, and it will be translated to:
+Compile the workflow `test` from above with the `-inputs mymanifest.json` option. A new file `mymanifest.dx.json` will be 
+created with the following content. **NOTE** `mymanifest.dx.json` is created by the compiler - the user does not need to 
+create/change it manually.
 
 `mymanifest.dx.json`
 ```json
 {
   "input_manifest___": {
-    "s": "hello",
-    "f": {
-      "$dnanexus_link": "file-xxx"
-    }
-  },
-  "input_manifest___files": [
-    {
-      "$dnanexus_link": "file-xxx"
+    "encoded": false,
+    "types": {
+      "f": "File",
+      "s": "String"
+    },
+    "values": {
+      "s": "hello",
+      "f": "project-aaa:file-xxx"
     }
-  ]
+  }
 }
 ```
 
-Finally, run your workflow using the translated input file:
-
-`dx run workflow-yyy -f mymanifest.dx.json`
+The created `mymanifest.dx.json` should be used as an input file when running the workflow:
+```commandline
+dx run workflow-yyy -f mymanifest.dx.json
+```
 
-## Manifest file
+#### Intermediate manifest file inputs/outputs
 
-Manifest files are less convenient to use as applet/workflow inputs because they must be uploaded to the platform. However, when manifest support is enabled, applet/workflow outputs are in the form of manifest files, so it is useful to understand the format.
+When manifest support is enabled, applet/workflow outputs which are passed from one stage to another (or to the final output 
+stage) exist on the form of intermediate manifests. Here we describe the format of such an intermediary manifest for users to better 
+understand the format. There is no need to use them as your workflow inputs, and JSON manifest above is the recommended way.
 
-Given the above workflow, the manifest output would be:
+Given the above workflow, the manifest output from `common` stage to the next stage (not shown) would be:
 
 ```json
 {
-  "id": "test",
+  "encoded": false,
+  "id": "stage-common",
   "values": {
-    "i": 1,
-    "p": {
-      "left": "hello",
-      "right": {
-        "$dnanexus_link": "file-xxx"
-      }
-    }
+    "s": "hello",
+    "f": "project-aaa:file-xxx"
   }
 }
 ```
 
-The `id` field is optional but will always be populated in the output manfiests. The manifest may contain additional fields (`types` and `definitions`) that are only for internal use and can be ignored.
+The `id` field represents the ID of the stage which created the manifest output. It is optional but will always be 
+populated in the output manifests. The manifest may contain additional fields (`types` and `definitions`) that are only 
+for internal use and can be ignored.
```

Comment:
does output stage produce a manfiest file?  Does user need to understand the format of the manifest file to interpret which output files correspond to which WDL workflow fields?

Reply:
Yes: https://github.com/dnanexus/dxCompiler/blob/deceb5969a512a847bb493bab77d8b6a0aa21fa4/doc/ExpertOptions.md?plain=1#L1447
Example:
`project-Fy9QqgQ0yzZbg9KXKP4Jz6Yqanalysis-GFBqG8j0yzZbjGyp159b46VK`. I elaborated further where the user should look for the outputs in the manifest file.